### PR TITLE
Gherkin Scripts for ODC-4181 and ODC-3954

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4835 and https://issues.redhat.com/browse/ODC-4860

Problem: Gherkin Scripts for scenarios in [ODC-4181](https://issues.redhat.com/browse/ODC-4181) and [ODC-3954](https://issues.redhat.com/browse/ODC-3954)

Solution: These are the gherkin scripts designed covering all the scenarios related to epics [ODC-4181](https://issues.redhat.com/browse/ODC-4181) and [ODC-3954](https://issues.redhat.com/browse/ODC-3954)

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Feature: Developer Catalog Page
+    As a user, I should be able to use Developer Catalog page to deploy application
+
+
+    Background:
+        Given user is at developer perspective
+        And user has installed OpenShift Serverless Operator
+        And user has installed Knative Apache Camelk Operator
+        And user has installed Apache Kafka Operator
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:9]: Why does user needs to install all these operators for using developer catalog? The catalog is available for all users without any operators installed.

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        And user will see Builder Images, Event Sources, Helm Charts, Operator Backed, Templates types
+        And user will see Filter by Keyword field
+        And user will A-Z, Z-A sort by dropdown
+
+    
+    @smoke, @manual
+    Scenario: Event Sources on Developer Catalog
+        Given user is at Developer Catalog page
+        When user clicks on Event Sources type
+        Then user will see the list of Providers
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:27]: This is not always true. User will see providers filter only if there are items from more than one provider in the catalog.

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        And user will see the cards of Event Sources
+        And user will see Filter by Keyword field
+        And user will A-Z, Z-A sort by dropdown
+
+
+    @smoke, @manual
+    Scenario: Helm Charts on Developer Catalog
+        Given user is at Developer Catalog page
+        When user clicks on Helm Charts type
+        Then user will see the list of Chart Repositories
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:37]: Same as above. Not always true.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature]: Updated.

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:25]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:36]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature [hunk 4]
```diff
@@ -11,7 +11,12 @@ Feature: Create event sources
    Scenario: Different event source enters display in event sources add page : Kn-07-TC03, Kn-08-TC02
       Given user is at Add page
       When user clicks on "Event Source" card
-      Then user will be redirected to page with header name "Event Sources"
+      Then user will get redirected to Developer Catalog page
+      And user will see header name "Event Sources"
+      And user will see the list of Providers
+      And user will see the cards of Event Sources
+      And user will see Filter by Keyword field
+      And user will A-Z, Z-A sort by dropdown
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature:19]: This is confusing. Is this what you mean here:
```suggestion
      And user will see A-Z, Z-A sort by dropdown
```

File: frontend/packages/dev-console/integration-tests/features/helm/install-HelmChart.feature [hunk 5]
```diff
@@ -17,15 +17,17 @@ Scenario: Developer Catalog Page when Helm Charts checkbox is selected: HR-01-TC
     Given user is at Add page
     When user clicks on the Helm Chart card on the Add page
     Then user will get redirected to Developer Catalog page
-    And user is able to see Helm Chart option is selected in Developer Catalog page
-    And user is able to see Helm Charts cards
+    And user will see the list of Chart Repositories
+    And user will see the cards of Helm Charts
+    And user will see Filter by Keyword field
+    And user will A-Z, Z-A sort by dropdown
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/helm/install-HelmChart.feature:23]:
```suggestion
    And user will see A-Z, Z-A sort by dropdown
```

File: frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/helm/install-HelmChart.feature (hunk 5 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/helm/install-HelmChart.feature:23]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        And user will see CI/CD, Databases, Languages, Middleware, Other categories
+        And user will see Builder Images, Event Sources, Helm Charts, Operator Backed, Templates types
+        And user will see Filter by Keyword field
+        And user will see A-Z, Z-A sort by dropdown
+
+    
+    @smoke, @manual
+    Scenario: Event Sources Providers on Developer Catalog
+        Given user is at Developer Catalog page
+        And user has installed event sources from other providers
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:23]: I think this will be more appropriate
```suggestion
        And user has installed event sources from different providers
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        Then user will see the list of Providers
+        And user will see the cards of Event Sources
+        And user will see Filter by Keyword field
+        And user will see A-Z, Z-A sort by dropdown
+
+
+    @smoke, @manual
+    Scenario: Helm Charts Repositories on Developer Catalog
+        Given user is at Developer Catalog page
+        And user has imported helm charts from other repositories
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:34]: Same as above
```suggestion
        And user has imported helm charts from different repositories
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:23]: updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:34]: updated

File: frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature [hunk 8]
```diff
@@ -11,7 +11,12 @@ Feature: Create event sources
    Scenario: Different event source enters display in event sources add page : Kn-07-TC03, Kn-08-TC02
       Given user is at Add page
       When user clicks on "Event Source" card
-      Then user will be redirected to page with header name "Event Sources"
+      Then user will get redirected to Developer Catalog page
+      And user will see header name "Event Sources"
+      And user will see the list of Providers
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature:16]: @gajanan-more will the same logic applies here that the event sources will be installed from different providers

Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature:16]: user will see list of providers only there are more than one provider type

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature [hunk 9]
```diff
@@ -0,0 +1,67 @@
+Feature: Event Sources Installation View
+    As a user, I should be able switch between YAML and Form view to install Event Sources
+
+    
+    Background:
+        Given user is at developer perspective
+        And user has installed OpenShift Serverless Operator
+        And user has installed Knative Apache Camelk Operator
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:8]: Camel K Integration operator

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        And user has installed Apache Kafka Operator
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:9]: why is this needed? Kafka is part of serverless operator itself , need to create CR for kafka

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        And user selects Resource radio button
+        And user selects "hello-openshift" as a resource
+        And user clicks on the Create button
+        Then user will be redirected to Topology page
+        And Topology page have the Event Source created
+
+
+    @regression, @smoke, @manual
+    Scenario: Switch from YAML to Form view
+        Given user is at the Create Event Source page
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:44]: what you mean here `Create Event Source page` there is no dedicated page now as it needs context based on what's selected from catalog flow for eventSources

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        When user selects the YAML View
+        And user does some changes in the yaml for Event Source
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:46]:
```suggestion
        And user does some valid changes in the yaml for Event Source
```

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+        And user selects the Form view
+        Then user will see that the data hasn't lost
+
+
+    @regression, @smoke, @manual
+    Scenario: Switch from Form to YAML view
+        Given user is at the Install Event Source page
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:53]: terminology used above is Create and see above comment

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources.feature [hunk 14]
```diff
@@ -10,13 +10,6 @@ Feature: Event Sources actions
...
-    @manual
-    Scenario: Scroll bar on the Event Sources cards page
-        Given user is at the Event Sources page
-        And user has added more than 10 Event Sources
-        Then user will see the Scroll bar added to see all the card
-
-
     @manual
     Scenario: Filter the Event Sources
         Given user is at the Event Sources page
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/event-Sources.feature:15]: there is no eventSource page if contextual now based on selection from catalog flow

File: frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature (hunk 8 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature:16]: Added one given statement "user has installed event sources from different providers"

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature (hunk 9 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:8]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature (hunk 10 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:9]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature (hunk 11 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:44]: It's not dedicated page but where we fill the form or update the YAML, the header of the page is "Create Event Source"

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature (hunk 12 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:46]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature (hunk 13 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/event-Sources-InstallationView.feature:53]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/event-Sources.feature (hunk 14 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/event-Sources.feature:15]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature (hunk 8 above)
Comment (invincibleJai) [reply frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature:17]: just note: provider will be different only if sources provide that in CRDs for builtIn we take exception and show Red Hat as provider as CRD doesn't provide now.

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 2 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:25]: How is this resolved? It still says the same thing.

File: frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature (hunk 3 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/integration-tests/features/addFlow/developer-Catalog-Details.feature:36]: How is this resolved? It still says the same thing.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature [hunk 15]
```diff
@@ -7,11 +7,17 @@ Feature: Create event sources
       And user has selected namespace "aut-create-knative-event-source"
 
 
-   @smoke
+   @smoke, @manual
    Scenario: Different event source enters display in event sources add page : Kn-07-TC03, Kn-08-TC02
       Given user is at Add page
+      And user has installed event sources from different providers
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/integration-tests/features/Knative/create-event-sources.feature:13]:
```suggestion
      And user has installed one or more operators that contribute event sources 
```

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Review by rohitkrai03 — APPROVED** [review]
/lgtm
